plugins/modules/grafana_datasource.py

```diff
@@ -701,10 +711,19 @@ def _send_request(self, url, data=None, headers=None, method="GET"):
             return self._module.from_json(resp.read())
         self._module.fail_json(failed=True, msg="Grafana API answered with HTTP %d for url %s and data %s" % (status_code, url, data))
 
-    def switch_organisation(self, org_id):
+    def switch_organization(self, org_id):
         url = "/api/user/using/%d" % org_id
         response = self._send_request(url, headers=self.headers, method='POST')
 
+    def organization_by_name(self, org_name):
+        url = "/api/user/orgs"
+        organizations = self._send_request(url, headers=self.headers, method='GET')
+        for org in organizations:
+            if org['name'] == org_name:
+                return org['orgId']
```

Comment:
We could use next + list comprehension here, making it shorter and faster:
https://stackoverflow.com/questions/9014058/creating-a-python-list-comprehension-with-an-if-and-break

```suggestion
            org = next((org for org in organizations if org['name'] == org_name))
            if org:
                return org['orgId']
```